### PR TITLE
rustfmt

### PR DESCRIPTION
**Pull request by Munksgaard**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -59,13 +59,12 @@
...
 //! }
 //! ```
-
 #![cfg_attr(feature = "chan_select", feature(mpsc_select))]
 
 use std::marker;
 use std::thread::spawn;
 use std::mem::transmute;
-use std::sync::mpsc::{Sender, Receiver, channel};
+use std::sync::mpsc::{channel, Receiver, Sender};
```
Comment (laumann) [inline src/lib.rs]: rustfmt also prefers imported functions listed before structs?

File: src/lib.rs [hunk 2]
```diff
@@ -541,10 +533,11 @@ pub fn session_channel<P: HasDual>() -> (Chan<(), P>, Chan<(), P::Dual>) {
...
 pub fn connect<F1, F2, P>(srv: F1, cli: F2)
-    where F1: Fn(Chan<(), P>) + marker::Send + 'static,
-          F2: Fn(Chan<(), P::Dual>) + marker::Send,
-          P: HasDual + marker::Send + 'static,
-          <P as HasDual>::Dual: HasDual + marker::Send + 'static
+where
+    F1: Fn(Chan<(), P>) + marker::Send + 'static,
+    F2: Fn(Chan<(), P::Dual>) + marker::Send,
+    P: HasDual + marker::Send + 'static,
+    <P as HasDual>::Dual: HasDual + marker::Send + 'static,
```
Comment (laumann) [inline src/lib.rs]: This line can now be be `P::Dual: HasDual + marker::Send + 'static` (no need for the cast anymore)

File: src/lib.rs (hunk 1 above)
Comment (Munksgaard) [reply src/lib.rs]: It might just be that it orders them lexicographically in order to avoid ambiguity?

**Comment by Munksgaard** [comment]
With #34, shouldn't it be 0.3.0? After all, we've added new features.

File: src/lib.rs (hunk 1 above)
Comment (laumann) [reply src/lib.rs]: But then Receiver should come before Sender, right?

**Comment by laumann** [comment]
Sure! I just figured that we haven't modified the existing API functions, just extended the API itself, so maybe a minor version bump wouldn't be necessary.

File: src/lib.rs (hunk 1 above)
Comment (Munksgaard) [reply src/lib.rs]: Good point :smile:. I wonder why it choses this ordering then....

**Pull request merged**